### ISSUE TITLE
Error on nodemon server restart

### ISSUE THREAD

**Issue by johnernaut** [apollographql/apollo-engine-js#162]
I'm getting the following error when nodemon restarts my application due to a file change (using the latest version of engine):

```
[nodemon] starting `node index.js`
INFO[0000] Started HTTP server.                          address="[::]:3000"
INFO[0000] Engine proxy started.                         build=2018.02-156-g4bb06d64b version=1.0.2
[nodemon] restarting due to changes...
panic: close of closed channel

goroutine 81 [running]:
mdg/engine/engineproxy/tracing.(*tracesState).Close(0xc420324280, 0x0, 0xc4204a2fd0)
	/Users/glasser/Projects/Meteor/monorepo/go/src/mdg/engine/engineproxy/tracing/traces.go:119 +0x44
mdg/engine/engineproxy/tracing.(*Reports).Close.func2(0xc42038c2a0, 0xc4204bc230)
	/Users/glasser/Projects/Meteor/monorepo/go/src/mdg/engine/engineproxy/tracing/reports.go:175 +0x34
created by mdg/engine/engineproxy/tracing.(*Reports).Close
	/Users/glasser/Projects/Meteor/monorepo/go/src/mdg/engine/engineproxy/tracing/reports.go:174 +0x113
[nodemon] starting `node index.js`
INFO[0000] Started HTTP server.                          address="[::]:3000"
INFO[0000] Engine proxy started.                         build=2018.02-156-g4bb06d64b version=1.0.2

```

I've made a bare-bones repo that reproduces this problem which can be found here:  [https://github.com/johnernaut/apollo-engine-error](https://github.com/johnernaut/apollo-engine-error).

I'd love to help submit a patch for this if given some direction on where to start, although I assume this may be an issue with the Go daemon.

**Comment by glasser** [comment]
Thanks for the good reproduction. I hadn't seen this when testing the new restart logic because I disabled reporting when testing and this is in the reporting logic.

I see the bug and have a fix for it under review internally. Will get it out soon though some jury duty may get in the way of that.

This isn't likely to actually cause any issues in development other than printing annoying messages, and is unlikely to occur in production as it is specific to SIGUSR2.

**Comment by johnernaut** [comment]
@glasser Gotcha - thanks for the info!

**Comment by glasser** [comment]
Fixed in 1.0.5. https://www.apollographql.com/docs/engine/proxy-release-notes.html

**Comment by axe-z** [comment]
I 've got that exact same issue, on the bleeding edge version.
